lib/rules/declaration-block-no-redundant-longhand-properties/index.js

```diff
@@ -113,21 +154,24 @@ const rule = (primary, secondaryOptions, context) => {
 							const transformedDeclarationNodes = new Map(
 								declNodes.map((d) => [d.prop.toLowerCase(), d]),
 							);
-							const resolvedShorthandValue = prefixedShorthandData
-								.map((p) => transformedDeclarationNodes.get(p)?.value.trim())
-								.filter(Boolean)
-								.join(' ');
+							const resolvedShorthandValue = resolveShorthandValue(
+								prefixedShorthandProperty,
+								prefixedShorthandData,
+								transformedDeclarationNodes,
+							);
 
-							const newShorthandDeclarationNode = firstDeclNode.clone({
-								prop: prefixedShorthandProperty,
-								value: resolvedShorthandValue,
-							});
+							if (resolvedShorthandValue !== null) {
```

Comment:
[suggest] This is just my preference, simply checking just a variable seems a bit more readable:

```suggestion
							if (resolvedShorthandValue) {
```